### PR TITLE
Update darc documentation

### PR DESCRIPTION
**Pull request by mmitche**
- Update documentation for add-build-to-channel
- Add docs for how to assign builds to channels
- Add docs for locating the BAR build ID.

### CODE REVIEW

File: Documentation/Darc.md [hunk 1]
```diff
@@ -793,6 +795,80 @@ The root build can then be provided using --id
...
+The default use case is shown below. The ID is the BAR build ID, which can be found in the Publish Build Assets
+task of the "Publish to Build Asset Registry" job in your build. See
+[Locating the BAR build ID for a build](#locating-the-bar-build-id-for-a-build).
+
+```
+darc add-build-to-channel --id 46856 --channel ".NET Core SDK 3.1.3xx"
+```
+
+There are a few interesting non-standard scenarios:
+- If you don't want to publish you can pass: `--skip-assets-publishing`. The build will immediately apply
```
Comment (riarenas) [inline Documentation/Darc.md:827]: Maybe add a note that this should only be done if you're sure the packages required by any downstream repos are already published before using this switch?

File: Documentation/Darc.md [hunk 2]
```diff
@@ -793,6 +795,80 @@ The root build can then be provided using --id
...
+  to the new channel.
+- By default, the publishing pipeline will use the same version of arcade that was referenced in the build
+  you wish to assign. This may not be desirable if your repo is missing a required arcade fix. You can
```
Comment (riarenas) [inline Documentation/Darc.md:830]: I think you missed a word here

File: Documentation/Darc.md [hunk 3]
```diff
@@ -793,6 +795,80 @@ The root build can then be provided using --id
...
+  for the automation to use a different arcade version by passing `--source-branch` and `--source-sha`. These
+  are the source branch and sha of arcade, not your repo. For instance, the following command will use arcade
+  at sha 09bb9d929120b402348c9a0e9c8c951e824059aa in context of branch master (this is required by AzDO, but not
```
Comment (riarenas) [inline Documentation/Darc.md:833]:
```suggestion
  at sha `09bb9d929120b402348c9a0e9c8c951e824059aa` in context of branch master (this is required by AzDO, but not
```

File: Documentation/Darc.md [hunk 4]
```diff
@@ -793,6 +795,80 @@ The root build can then be provided using --id
...
+        47380, IsProduct: True
+        47391, IsProduct: True
+        41865, IsProduct: True
+        47632, IsProduct: False
+    Metadata has been pushed. Build id in the Build Asset Registry is '47814'
+    Found the following default channels:
+        https://github.com/dotnet/installer@master => (131) .NET Core 5 Dev
+    Determined build will be added to the following channels: [131]
+  ```
+
```
Comment (riarenas) [inline Documentation/Darc.md:871]: [nit] Maybe add a conclusion to say, in this case, the build ID is '47814' ?

**Comment by riarenas** [comment]
This fixes https://github.com/dotnet/arcade/issues/5159

File: Documentation/Darc.md [hunk 5]
```diff
@@ -1170,6 +1246,18 @@ parameters:
...
 - `--channel` - **(Required)**. Channel to assign build to.
+- `--source-branch` - Branch that should be used as base for the promotion build.
+- `--source-sha` - SHA that should be used as base for the promotion build.
+- `--validate-signing` - Perform signing validation.
+- `--validate-nuget` - Perform NuGet metadata validation.
+- `--validate-sourcelink` - Perform SourceLink validation.
+- `--validate-SDL` - Perform SDL validation.
+- `--sdl-validation-parameters` - Custom parameters for SDL validation.
+- `--sdl-validation-continue-on-error` - Ignore SDL validation errors.
+- `--skip-assets-publishing` - Add the build to the channel without publishing assets to th
```
Comment (JohnTortugo) [inline Documentation/Darc.md:1257]: NIT: "thE"

File: Documentation/Darc.md [hunk 6]
```diff
@@ -793,6 +795,80 @@ The root build can then be provided using --id
...
+  for the automation to use a different arcade version by passing `--source-branch` and `--source-sha`. These
+  are the source branch and sha of arcade, not your repo. For instance, the following command will use arcade
+  at sha 09bb9d929120b402348c9a0e9c8c951e824059aa in context of branch master (this is required by AzDO, but not
+  particularly material to the functionality.)
+  ```
+  darc add-build-to-channel --id 46856 --channel ".NET Core SDK 3.1.3xx" --source-branch master
+    --source-sha 09bb9d929120b402348c9a0e9c8c951e824059aa
+  ```
+
+### Locating the BAR build ID for a build
```
Comment (JohnTortugo) [inline Documentation/Darc.md:840]: I wish there were an easier way to find this value..

Comment (mmitche) [reply Documentation/Darc.md:840]: It's coming in my next PR.

**Pull request merged**